python/ray/serve/controller.py

```diff
@@ -289,7 +292,8 @@ def _update_active_nodes(self):
         replicas). If the active nodes set changes, it will notify the long poll client.
         """
         new_active_nodes = self.deployment_state_manager.get_active_node_ids()
-        new_active_nodes.add(self._head_node_id)
+        head_node_id = ray.get_runtime_context().node_id
+        new_active_nodes.add(head_node_id)
```

Comment:
this can't change through the lifetime of the controller, so do it once in constructor (else this is a call to GCS every update loop)